core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderStub.java

```diff
@@ -1,19 +1,34 @@
 package nl.nn.adapterframework.http.cxf;
 
-import nl.nn.adapterframework.core.PipeLineSession;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.SneakyThrows;
 import nl.nn.adapterframework.core.ListenerException;
+import nl.nn.adapterframework.core.PipeLineSession;
 import nl.nn.adapterframework.stream.Message;
 
 public class SoapProviderStub extends SOAPProviderBase {
 
 	PipeLineSession session = null;
 
+	final Map<String, Message> sessionCopy = new HashMap<>();
+
 	@Override
+	@SneakyThrows
 	Message processRequest(String correlationId, Message message, PipeLineSession pipelineSession) throws ListenerException {
-		if(session != null)
+		if(session != null) {
 			pipelineSession.putAll(session);
-
+			session.getCloseables().clear();
+		}
 		session = pipelineSession;
+		for (String key : session.keySet()) {
+			if (!(session.get(key) instanceof InputStream)) {
+				// Do not copy input-streams b/c then they are no longer readable
+				sessionCopy.put(key, session.getMessage(key).copyMessage());
+			}
+		}
```

Comment:
Ik snap dit niet helemaal maar het is een (oude) test..

Reply:
> Ik snap dit niet helemaal maar het is een (oude) test..

De test probeert messages te lezen die in de session zitten, maar omdat de session gesloten is zijn die messages nu NULL.
Dus kopieer ik de session inhoud en maak een kopie van iedere session value.

Alleen niet als het een input stream is want dan lees je die stream en ben je hem kwijt. (Er is 1 test die een input stream in de session doet).

Ik had misschien ook alleen de instance-of-message values kunnen kopiëren.